certbot/compat.py

```diff
@@ -0,0 +1,104 @@
+"""
+Compatibility layer to run certbot both on Linux and Windows.
+
+The approach used here is similar to Modernizr for Web browsers.
+We do not check the plateform type to determine if a particular logic is supported.
+Instead, we apply a logic, and then fallback to another logic if first logic
+is not supported at runtime.
+
+Then logic chains are abstracted into single functions to be exposed to certbot.
+"""
+import os
+import select
+import sys
+import errno
+import ctypes
+
+from certbot import errors
+
+try:
+    # Linux specific
+    import fcntl # pylint: disable=import-error
+except ImportError:
+    # Windows specific
+    import msvcrt # pylint: disable=import-error
+
+def raise_for_non_administrative_windows_rights():
+    """
+    On Windows, raise if current shell does not have the administrative rights.
+    Do nothing on Linux.
+    """
+    if hasattr(ctypes, 'windll'):
+        windll = getattr(ctypes, 'windll')
+        if windll.shell32.IsUserAnAdmin() == 0:
```

Comment:
Because the static code analysis become insane at this point, it does not accept some code manipulation on modules where the object exposed depends on the runtime platform. Here, the windll object on the ctype module, that does not exists on a Linux platform.

So to solve that, I was forced to hide the existence or non-existence of the object to the static code analysis by reflection.